packages/binding-http/src/oauth-token-validation.ts

```diff
@@ -60,11 +60,11 @@ function extractTokenFromRequest(request: http.IncomingMessage) {
     const url = new URL(request.url ?? "", `http://${request.headers.host}`);
     const queryToken = url.searchParams.get("access_token");
 
-    if (!headerToken && !queryToken) {
+    if (headerToken == null && queryToken == null) {
         throw new Error("Invalid request: only one authentication method is allowed");
```

Comment:
Somehow the error message does not seem right.. or confuses my.. the code checks whether both are null.. but the error says just one method is allowed?

Reply:
Oh, good point! Seems like the logic was intended to check for the exact opposite, right (i.e., `headerToken != null && queryToken != null`)?

Reply:
I think so too but let's wait for @relu91 to comment...

Reply:
yes, exactly the translation is wrong. It should be as [suggested](https://github.com/eclipse-thingweb/node-wot/pull/1097/files#r1335868880).

Reply:
Since it is more than a refactoring but a change of the actual logic, should we maybe make this change via a separate PR? Or is fine to keep it in here?

Reply:
I think in this case I am fine with integrating it as part of this PR also. But either way is fine..

Reply:
Okay, then we should be ready to merge this one :)